### PR TITLE
browsers-w1-homework

### PR DESCRIPTION
**Pull request by SaidYein**

### CODE REVIEW

File: 2-Browsers/Week1/homework/ex1-bookList/index.js [hunk 1]
```diff
@@ -39,9 +39,54 @@ const myBooks = [
...
+  function renderProductList(item) {
+    // create list items and give them a className
+    const li = document.createElement('li');
+    li.setAttribute('class', 'item');
+
+    //wrap the the title and the author by <p>
+    const p = document.createElement('p');
+    p.textContent = item.title + ' - ' + item.author;
+    li.append(p);
+    //add some style (doing style here just for practicing DOM. Otherwise, I would create style tag in HTML or do the styling in CSS file)
```
Comment (cgduncan7) [inline 2-Browsers/Week1/homework/ex1-bookList/index.js:61]: 👍

File: 2-Browsers/Week1/homework/ex1-bookList/index.js [hunk 2]
```diff
@@ -39,9 +39,54 @@ const myBooks = [
...
+  const ul = document.createElement('ul');
+  // Give id tu the list
+  ul.setAttribute('id', 'list');
+  ul.style.display = 'inline-flex';
+
+  //For each item in the -books-, call the function that creates html piece
+  books.forEach(renderProductList);
+
+  // the function wraps the items in the object by html tags:
+  function renderProductList(item) {
```
Comment (cgduncan7) [inline 2-Browsers/Week1/homework/ex1-bookList/index.js:52]: You can have this function defined outside of `createBookList` function

File: 2-Browsers/Week1/homework/ex1-bookList/index.js [hunk 3]
```diff
@@ -39,9 +39,54 @@ const myBooks = [
...
 
 const ulElement = createBookList(myBooks);
 
 document.querySelector('#bookList').appendChild(ulElement);
+
+// creating <img> tags with particular src and inserting them to li elements -which are child of 'ul'- one by one
+const img_1 = document.createElement('img');
+img_1.setAttribute('src', 'assets/the_design_of_everyday_things.jpg');
+img_1.setAttribute('alt', 'book cover')
+ulElement.children[0].appendChild(img_1);
```
Comment (cgduncan7) [inline 2-Browsers/Week1/homework/ex1-bookList/index.js:82]: This should probably be done within `renderProductList`.

File: 2-Browsers/Week1/homework/ex1-bookList/index.js [hunk 4]
```diff
@@ -39,9 +39,54 @@ const myBooks = [
...
 }
 
 const ulElement = createBookList(myBooks);
 
 document.querySelector('#bookList').appendChild(ulElement);
+
+// creating <img> tags with particular src and inserting them to li elements -which are child of 'ul'- one by one
+const img_1 = document.createElement('img');
+img_1.setAttribute('src', 'assets/the_design_of_everyday_things.jpg');
+img_1.setAttribute('alt', 'book cover')
```
Comment (cgduncan7) [inline 2-Browsers/Week1/homework/ex1-bookList/index.js:81]: If the `alt` is always the same why not make a reusable function to create your custom `img` elements and then use that?

Comment (cgduncan7) [reply 2-Browsers/Week1/homework/ex1-bookList/index.js:81]: Also, adding to DOM after it has been rendered will be a bit costly. This solution also only works for 3 books!

File: 2-Browsers/Week1/homework/ex2-aboutMe/index.js [hunk 5]
```diff
@@ -10,4 +10,11 @@ Full description at: https://github.com/HackYourFuture/Homework/tree/main/2-Brow
    `list-item`.
 ------------------------------------------------------------------------------*/
 
-// TODO add your JavaScript code here.
+document.body.style.fontFamily = 'Arial, sans-serif';
+const list = document.querySelectorAll('li');
+const aboutMe = ['SaidYein', 'Pizza', 'Istanbul'];
+
+for (let i = 0; i < list.length; i++) {
+  list[i].textContent = aboutMe[i];
```
Comment (cgduncan7) [inline 2-Browsers/Week1/homework/ex2-aboutMe/index.js:18]: You need to change the content of the `span` within this `li`. Not the `li` itself.

File: 2-Browsers/Week1/homework/ex2-aboutMe/index.js [hunk 6]
```diff
@@ -10,4 +10,11 @@ Full description at: https://github.com/HackYourFuture/Homework/tree/main/2-Brow
    `list-item`.
 ------------------------------------------------------------------------------*/
 
-// TODO add your JavaScript code here.
+document.body.style.fontFamily = 'Arial, sans-serif';
```
Comment (cgduncan7) [inline 2-Browsers/Week1/homework/ex2-aboutMe/index.js:13]: You forgot to do step 4!

File: 2-Browsers/Week1/homework/ex2-aboutMe/index.js [hunk 7]
```diff
@@ -10,4 +10,11 @@ Full description at: https://github.com/HackYourFuture/Homework/tree/main/2-Brow
...
+const list = document.querySelectorAll('li');
+const aboutMe = ['SaidYein', 'Pizza', 'Istanbul'];
+
+for (let i = 0; i < list.length; i++) {
+  list[i].textContent = aboutMe[i];
+  list[i].setAttribute('class', 'list-item');
```
Comment (cgduncan7) [inline 2-Browsers/Week1/homework/ex2-aboutMe/index.js:19]: 👍

File: 2-Browsers/Week1/homework/ex4-whatsTheTime/index.js [hunk 8]
```diff
@@ -8,7 +8,21 @@ Full description at: https://github.com/HackYourFuture/Homework/tree/main/2-Brow
...
+  const body = document.body;
+
+  //some unnecessary styling :)
+  body.style.display = 'flex';
+  body.style.justifyContent = 'center';
+  body.style.marginTop = '25%';
+  body.style.fontSize = '8rem';
+
+  //getting time in HH:MM:SS
+  const dateTime = currentDate.toLocaleTimeString();
```
Comment (cgduncan7) [inline 2-Browsers/Week1/homework/ex4-whatsTheTime/index.js:21]: 👏 don't reinvent the wheel 😆

File: 2-Browsers/Week1/homework/ex4-whatsTheTime/index.js [hunk 9]
```diff
@@ -8,7 +8,21 @@ Full description at: https://github.com/HackYourFuture/Homework/tree/main/2-Brow
...
+  body.textContent = dateTime;
 }
+window.onload = function () {
+  setInterval(addCurrentTime, 1000);
+};
 
-// TODO execute `addCurrentTime` when the browser has completed loading the page
+//PS: this is my second version. I did HH:MM:SS format with function manually without using 'toLocaleTimeString()' in the first version. I like that one better but this is cleaner :)
```
Comment (cgduncan7) [inline 2-Browsers/Week1/homework/ex4-whatsTheTime/index.js:28]: 👍

File: 2-Browsers/Week1/homework/ex5-catWalk/index.js [hunk 10]
```diff
@@ -21,8 +21,43 @@ Full description at: https://github.com/HackYourFuture/Homework/tree/main/2-Brow
 
    https://media1.tenor.com/images/2de63e950fb254920054f9bd081e8157/tenor.gif
 -----------------------------------------------------------------------------*/
+const img = document.querySelector('img');
+img.style.left = '0px';
+const halfway = (window.innerWidth - img.width) / 2;
+const dancingCat =
+  'https://media1.tenor.com/images/2de63e950fb254920054f9bd081e8157/tenor.gif';
+const walkingCat = 'http://www.anniemation.com/clip_art/images/cat-walk.gif';
```
Comment (cgduncan7) [inline 2-Browsers/Week1/homework/ex5-catWalk/index.js:29]: well done having a `const`-section. Makes it easier to reuse these variables throughout your code

File: 2-Browsers/Week1/homework/ex5-catWalk/index.js [hunk 11]
```diff
@@ -21,8 +21,43 @@ Full description at: https://github.com/HackYourFuture/Homework/tree/main/2-Brow
...
+    } else {
+      // I let the action start -300 px, because the width og gif is 296. Transactions seems smoother this way
+      step = -300;
+    }
+    //adding the distance stored in step to the left of the gif
+    img.style.left = step + 'px';
+  }
+
+  //I have designed the action in to three parts: before halfway, halfway and after halfway, So below will call 'step function after and before halfway'
+  if (step === halfway - (halfway % 10)) {
```
Comment (cgduncan7) [inline 2-Browsers/Week1/homework/ex5-catWalk/index.js:48]: Smart way to make sure that `halfway` is always reached!

File: 2-Browsers/Week1/homework/ex5-catWalk/index.js [hunk 12]
```diff
@@ -21,8 +21,43 @@ Full description at: https://github.com/HackYourFuture/Homework/tree/main/2-Brow
...
+      }
+    } else {
+      // I let the action start -300 px, because the width og gif is 296. Transactions seems smoother this way
+      step = -300;
+    }
+    //adding the distance stored in step to the left of the gif
+    img.style.left = step + 'px';
+  }
+
+  //I have designed the action in to three parts: before halfway, halfway and after halfway, So below will call 'step function after and before halfway'
```
Comment (cgduncan7) [inline 2-Browsers/Week1/homework/ex5-catWalk/index.js:47]: Nice clean code 👍

**Review by cgduncan7 — CHANGES_REQUESTED** [review]
Really nicely done. Comments and code organization are great 👍 

Take a look at ex2, you forgot one step and did one step incorrectly!

**Comment by cgduncan7** [comment]
Also, you can include your unit-test reports as well 😄

File: 2-Browsers/Week1/homework/ex1-bookList/index.js [hunk 13]
```diff
@@ -19,74 +19,74 @@ https: //hyf-js2-week1-makeme-ex1-demo.herokuapp.com/
```
Comment (SaidYein) [inline 2-Browsers/Week1/homework/ex1-bookList/index.js:19]: I am not sure if I was allowed to change myBooks object but I needed indexes and could not do what I wanted without changing the object.

File: 2-Browsers/Week1/homework/ex1-bookList/index.js [hunk 14]
```diff
@@ -19,74 +19,74 @@ https: //hyf-js2-week1-makeme-ex1-demo.herokuapp.com/
...
+  li.append(img);
 
-const img_2 = document.createElement('img');
-img_2.setAttribute('src', 'assets/the_most_human_human.jpg');
-img_2.setAttribute('alt', 'book cover')
-ulElement.children[1].appendChild(img_2);
+  ul.appendChild(li);
+}
+// creating Image Element
+function createImageElement(index) {
```
Comment (cgduncan7) [inline 2-Browsers/Week1/homework/ex1-bookList/index.js:84]: Nice 💯

**Review by cgduncan7 — APPROVED** [review]
Much better, nice improvements @SaidYein 👏

**Pull request closed without merging**